### PR TITLE
#164798258 Users should be able to comment on articles.

### PR DESCRIPTION
**Pull request by anyatibrian**
### What does this PR do?

- Creates functionality to enable users comment on an article
### Description of tasks to be completed

- add test for the comments feature
- add respective UI designs
- add actions for the feature
- add reducers for the action
- add the comments components to the feature

### How can this be tested?

- Clone the repo
- Checkout ft-article-comments-164798258
- npm install && npm start
- log in with your account credentials and get redirected to the homepage upon successful login.
On the home page, click on one of the articles to full view it. On the articles-view page, below the body of the article there is a form field  from which you can enter your comments

### Relevant PT board story
[#164798258](https://www.pivotaltracker.com/story/show/164798258)

### Screen shots
Screenshot 2019-05-23 at 15 56 05

### CODE REVIEW

**Review by e-ian — CHANGES_REQUESTED** [review]
@anyatibrian your PR titile should begin with the story id and the commit message shouldnot have capital letters except the [Finishes #1612345] part

File: src/components/comments.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    const data = {
+      body: comments
+    };
+    articleComments(data, articleSlug);
+    fetchComments(articleSlug);
+  };
+
+  render() {
+    const { commentsMessage } = this.props;
+    console.log("comments", commentsMessage);
```
Comment (e-ian) [inline src/components/comments.js]: Remove console logs from the code

File: src/reducers/commentsReducer.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { CREATE_COMMENTS, CREATE_COMMENTS_ERRORS } from "../actions/types";
+
+const initialState = {
+  success: {},
+  errors: {}
+};
+
+export default (state = initialState, action) => {
+  console.log("reducer");
```
Comment (e-ian) [inline src/reducers/commentsReducer.js]: Please remove these console logs

File: src/styles/comments.scss [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  textarea.form-control {
+    border: 1px solid #e5e5e5;
+    height: 100px;
+  }
+
+  .btn-comment {
+    border: 1px solid #ccc;
+    float: right;
+  }
+}
```
Comment (e-ian) [inline src/styles/comments.scss]: Add a line at the end of this line

File: src/actions/commentsActions.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        payload: res.data
+      });
+      toast.dismiss();
+      toast.success("thanks for your view", {
+        hideProgressBar: false,
+        autoClose: 3000
+      });
+    })
+    .catch(errors => {
+      console.log(errors.response.data);
```
Comment (Genza999) [inline src/actions/commentsActions.js]: Hey @anyatibrian , great work. Kindly remove the console.logs in your code.

**Review by Genza999 — APPROVED** [review]

**Review by patrickf949 — APPROVED** [review]
Great Implementation @anyatibrian LGTM

**Review by e-ian — APPROVED** [review]
LGTM

File: src/components/comments.js (hunk 1 above)
Comment (anyatibrian) [reply src/components/comments.js]: done

File: src/actions/commentsActions.js (hunk 4 above)
Comment (anyatibrian) [reply src/actions/commentsActions.js]: done

File: src/reducers/commentsReducer.js (hunk 2 above)
Comment (MuhanguziDavid) [reply src/reducers/commentsReducer.js]: @anyatibrian , please remove the `console.log` statement.

Comment (anyatibrian) [reply src/reducers/commentsReducer.js]: done

File: src/styles/comments.scss (hunk 3 above)
Comment (anyatibrian) [reply src/styles/comments.scss]: done

**Review by MuhanguziDavid — COMMENTED** [review]
Nice work on comments @anyatibrian , please add timestamps for each comment as stated in the acceptance criteria for the comments story

Clicking the name of the user who posted a comment redirects to the articles page, please change this to redirect to the user's profile.

**Comment by anyatibrian** [comment]
> Nice work on comments @anyatibrian , please add timestamps for each comment as stated in the acceptance criteria for the comments story
> 
> Clicking the name of the user who posted a comment redirects to the articles page, please change this to redirect to the user's profile.

all done you can check it out

**Pull request merged**